lib/broccoli/ember-app.js

```diff
@@ -335,7 +335,7 @@ EmberApp.prototype._initVendorFiles = function() {
   }
 
   // don't crash if ember-cli-shims is not included
-  if (!bowerDeps['ember-cli-shims']) {
+  if (!ember && !bowerDeps['ember-cli-shims']) {
```

Comment:
Okay, so the _only_ thing which requires this change is `[ember-new-modules-shim](https://github.com/DockYard/ember-new-modules-shim)`. The users of that addon should be assumed to be highly familiar Ember devs.

Given that, if users of `ember-new-modules-shim` wish to not include `ember-cli-shims` they will need to follow the _full_ set of instructions at https://github.com/DockYard/ember-new-modules-shim#optional-disabling-ember-cli-shims ... not just remove the `bower` dependency and rely on this change.

That is the only way we can make sure that shims are not included across both `ember-source`-provided and `bower`-provided shims while knowing user intent.